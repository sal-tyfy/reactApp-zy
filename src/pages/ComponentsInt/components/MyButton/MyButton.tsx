export const MyButton = ({ children }: { children: React.ReactNode }) => {
  return (
    <button type="button" className="btn">
      {children}
    </button>
  );
};
