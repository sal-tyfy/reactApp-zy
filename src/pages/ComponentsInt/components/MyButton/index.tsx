import { Card } from 'antd';
import { MyButton } from './MyButton';

export const MyButtonWrapper = () => {
  return (
    <Card title="MyButton">
      <MyButton>click</MyButton>
    </Card>
  );
};
