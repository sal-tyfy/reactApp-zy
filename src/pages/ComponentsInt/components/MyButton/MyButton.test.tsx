import { render } from '@testing-library/react';
import { MyButton } from './MyButton';
import renderer from 'react-test-renderer';

describe('test MyButton', () => {
  it('should include text', () => {
    const { container } = render(<MyButton>click</MyButton>);
    const com = container.firstChild as HTMLElement;
    expect(com.tagName).toBe('BUTTON');
    expect(com.textContent).toBe('click');
  });
  it('should match snapshot', () => {
    const btn = renderer.create(<MyButton>click</MyButton>).toJSON();
    expect(btn).toMatchSnapshot();
  });
});
