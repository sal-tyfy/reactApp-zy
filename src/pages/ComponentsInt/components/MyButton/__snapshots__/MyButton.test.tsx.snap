// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyButton should match snapshot 1`] = `
<button
  className="btn"
  type="button"
>
  click
</button>
`;
